File: zigzag.js
var convert = function(s, numRows) {
  let arrs = [];
  for(let i = 0; i<numRows; i++){
    arrs.push([])
  }
  let counter = -1, up = true, down = false;
  for(let i = 0; i<s.length; i++){
    if(counter < 0){
      up = true;
      down = false;
    } 
    if(up) counter++;
    if(counter === numRows){
      up = false;
      down = true;
    }
    if(down) counter--;
    arrs[i].push(s[i]);
  }
  let op = '';
  for(let i = 0; i<numRows; i++){
    op += arrs[i].join('')
  }
  return op
};

File: zigzag.py
def convert(self, s: str, numRows: int) -> str:
    if numRows == 1:
        return s
    arrays = [[] for n in range(numRows)]
    down = True
    row = 0
    for c in s:
        arrays[row].append(c)
        if down:
            row += 1
            if row == numRows - 1:
                down = False
        else:
            row -= 1
            if row == 0:
                down = True
    string = ''
    for i in arrays:
        string += ''.join(i)
    return string